src/components/Button.tsx

```diff
@@ -66,11 +67,27 @@ export const StartButton: FunctionComponent<StartButtonProps> = ({
 
 export const BackButton: FunctionComponent<BackButtonProps> = ({
   href,
-}: BackButtonProps): JSX.Element => (
-  <a href={href} className="govuk-back-link">
-    Back
-  </a>
-);
+}: BackButtonProps): JSX.Element => {
+  return (
+    <Link href={href}>
+      <a className="govuk-back-link">Back</a>
+    </Link>
+  );
+};
+
+export const BackButtonRouterIndexes: FunctionComponent<BackButtonProps> = ({
+  href,
+}: BackButtonProps): JSX.Element => {
+  const router = useRouter();
+  const useIndexValue = router?.query.useIndex || 0;
+  href = `${href}?useIndex=${useIndexValue}`;
```

Comment:
@matthew-a-carr Extract `formatDestinationUrl` method to concatenate the query param to the user provided, accounting for query params (?) already within the url provided